visualization/dashboard/src/MLIDashboard/Controls/CohortEditor/CohortEditor.tsx

```diff
@@ -372,14 +417,59 @@ export class CohortEditor extends React.PureComponent<ICohortEditorProps, ICohor
     }
 
     private setFilterLabel(filter: IFilter): React.ReactNode {
-        //TODO: change the function unwrap neatly 
+        const selectedFilter = this.props.jointDataset.metaDict[filter.column];
         let label = "";
-        const space = " ";
-        label = this.props.jointDataset.metaDict[filter.column].abbridgedLabel + space
-        label += filter.method + space
-        label += filter.arg
+        label = selectedFilter.abbridgedLabel
+
+        const filterMethod = this.getFilterMethodLabel(filter.method)
+            label += filterMethod
+        if (selectedFilter.isCategorical) {
+            let selectedValues = [];
+            let filterArgs = filter.arg as number[];
+            filterArgs.forEach((element) => {
+                selectedValues.push(selectedFilter.sortedCategoricalValues[element])
+            }
+            );
+            label += selectedValues;
+        }
+        else {
+            label += filter.arg
+        }
+
+        if (filter.method == FilterMethods.inTheRangeOf) {
+            label += localization.FilterOperations.inTheRangeOfClose
+        }
+
+        return (<TooltipHost
+        overflowMode={TooltipOverflowMode.Self}
+        hostClassName={styles.filterLabel}
+        content={label}
+        onTooltipToggle={() => false}
+        styles={tooltip}
+      >
+        {label}
```

Comment:
do you need to set content above and here?

Reply:
Yes - content is used to display the overflow text in tooltip and {label} is used to display the label in the TooltipHost (TooltipHost is like <text> or <label>)
See `Tooltip only on overflow` in https://developer.microsoft.com/en-us/fluentui#/controls/web/tooltip